test/python/result/test_marginal_counts.py

```diff
@@ -0,0 +1,39 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=missing-class-docstring,missing-function-docstring
+
+"""Test marginal_counts function."""
+
+import unittest
+
+from qiskit.result.utils import marginal_counts
+
+
+class TestMarginalCounts(unittest.TestCase):
+    def test_default(self):
```

Comment:
Thanks for the tests @mattwright99, but I actually meant could you add a test for the case that threw the error previously (I believe it was the `indices=None` case?). Sorry if I wasn't clear before!

Also it looks like CI is failing because of linting issues. You can run `tox -eblack` to fix

Reply:
Oh I see, sorry. Since `indices=None` is the default setting, it is tested in the very first function `test_default`. Would you like me to be more explicit?

Reply:
When I run your default test against main I don't get any error, so it doesn't seem to be covering the edge case of the original error. I think it only crops up when you pass a result object, not just the raw counts

Reply:
Yes you are completely right, thanks for catching that. I will add a proper test now.

Reply:
Please could this test have a more descriptive name of what's being tested?  It's actually a test that passing `indices=None` succeeds - the fact that that's the default is not really important.